### ISSUE TITLE
Contrast functional patterns vs. OOP design patterns in README.md

### ISSUE THREAD

**Issue by ash-teach** [nic-dgl104-winter-2024/pattern-library#27]
**Note**: We won't discuss functional programming paradigm until week 12 so work on this issue may not materialize. However, there are at least a few students in the class with some experience in functional programming. Any student for whom functional programming is an interesting topic should feel encouraged to take this task on.

**The background**: Design patterns are (almost by definition) object-oriented and are exclusive to object-oriented programming. How are patterns like this represented in functional code? Are there comparable types of patterns? Or are they even necessary?

**The task**: Do some research into how the patterns presented in this repository translate (if at all) to a functional _style_ of programming may adopt patterns similar to those discussed in this repository; or, how functional programming may make adopting such patterns any of impossible, improbable, or unnecessary. Present some code that helps to explain the argument and include well-formatted references. 
- Unless you are already familiar with a pure functional language (such as [Haskell](https://www.haskell.org/), [Lean](https://lean-lang.org/), [Clojure](https://clojure.org/), [Racket](https://racket-lang.org/), or [Common Lisp](https://lisp-lang.org/), among others) you should concentrate on discussing functional programming _style_ in a language that has some functional capabilities, such as [JavaScript](https://learn-javascript.dev/docs/functional-programming), [Kotlin](https://docs.google.com/presentation/d/19C10TZM1kT0AzEjqSfLZs1_HC3Ye0E9h6muVDikl4uo/edit#slide=id.p1) or [Dart](https://dev.to/maximsaplin/embrace-functional-programming-with-dart-31-42cn).

**Comment by Sumit-tech-joshi** [comment]
Hi @ash-teach , I would like to work on this enhancement/documentation. Could you please assign it to me? I am choosing JavaScript functional programming to compare functional patterns vs. OOP design patterns, as it supports both paradigms.

I just want to know, is there any specific format that I should follow?

**Comment by ash-teach** [comment]
Sounds great. I would recommend doing some initial research into the questions posed above, and reporting your findings here. That will give others an opportunity to examine and discuss, and will ensure that we achieve the best possible outcome and direction. And using JavaScript is fine - though you should be careful to state that JavaScript is _not_ a pure functional language.

**Comment by Sumit-tech-joshi** [comment]
Sure

**Comment by Sumit-tech-joshi** [comment]
Hi @ash-teach, I have done some research on the topic and created a PR #59 for the same, Please let me know your view on this.

**Comment by mrsaif888** [comment]
hey @ash-teach  can work on this?

**Comment by mrsaif888** [comment]
hey @Sumit-tech-joshi can i join you with this task please

**Comment by ash-teach** [comment]
@mrsaif888 - I've requested some changes by @Sumit-tech-joshi in his PR #59. But if you would like to add additional examples you are welcome to do so (i.e. higher-order functions is one example - can you identify another example of a functional-style pattern?).

Once #59 is merged you can sync up and add some additional examples, if you like.

**Comment by Sumit-tech-joshi** [comment]
Hi @ash-teach, the PR #59 got closed due to merge conflicts. I have created a new PR with all requested changes in #73. Please review and let me know if anything needs to be added.

**Comment by mrsaif888** [comment]
@ash-teach  @Sumit-tech-joshi  i have add some more examples in PR #92  please review it

**Comment by ting-dev-coder** [comment]
closed this as completed in #73
